File: rb/21.rb
def divsum n
  (1..n / 2).select { |m| n % m == 0 }.reduce(:+)
end

sum = 0

(2..10000).each do |i|
  ds = divsum i
  sum += i if i == divsum(ds) && i != ds
end

p sum

File: js/21.js
function divsum(n) {
    var sum = 0;
    for (var i = 1; i <= n / 2; ++i)
        sum += n % i == 0 ? i : 0;
    return sum;
};

var sum = 0
for (var i = 2; i < 10000; ++i) {
    var ds = divsum(i);
    if (i == divsum(ds) && i != ds)
        sum += i;
}

console.log(sum);
